### PR TITLE
added tests for wp_check_filetype

### PR DESCRIPTION
**Pull request by pbearne**
Just tests

Trac ticket: [](https://core.trac.wordpress.org/ticket/57151)

### CODE REVIEW

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Test the wp_upload_bits function
+ *
+ * @group Functions
+ * @group Upload
+ * @covers ::wp_check_filetype
+ */
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:9]:
```suggestion
/**
 * Tests the wp_check_filetype() function.
 *
 * @group functions.php
 * @group upload
 * 
 * @covers ::wp_check_filetype
 */
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+class Test_wp_check_filetype extends WP_UnitTestCase {
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:10]:
```suggestion
class Tests_Functions_WpCheckFiletype extends WP_UnitTestCase {
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+	/**
+	 * @ticket 57151
+	 * @dataProvider wp_check_filetype_dataset
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:14]:
```suggestion
	 * @ticket 57151
	 *
	 * @dataProvider data_wp_check_filetype
	 *
	 * @param string     $filename The filename to check.
	 * @param array|null $mimes    An array of mimetypes.
	 * @param array      $expected An array containing the extension and mimetype.
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	 */
+	function test_wp_check_filetypes( $filename, $mines, $expected ) {
+		$this->assertSame( $expected, wp_check_filetype( $filename, $mines ) );
+	}
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:18]:
```suggestion
	public function test_wp_check_filetype( $filename, $mimes, $expected ) {
		$this->assertSame( $expected, wp_check_filetype( $filename, $mimes ) );
	}
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+	function wp_check_filetype_dataset() {
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:20]:
```suggestion
	/**
	 * Data provider for test_wp_check_filetype().
	 *
	 * @return array[]
	 */
	public function data_wp_check_filetype() {
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+		return array(
+			'default'     => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:22]:
```suggestion
			'.jpg and default allowed mimetypes' => array(
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+				'filename' => 'canola.jpg',
+				'mines'    => null,
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:24]:
```suggestion
				'mimes'    => null,
```
While this dataset covers the defaults (i.e. `null`), I'd also suggest an additional dataset that sets `$mimes` to `array()` to reflect an `empty()` value of the documented datatype.

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+				'expected' => array(
+					'ext'  => 'jpg',
+					'type' => 'image/jpeg',
+				),
+			),
+			'short_mines' => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:30]:
```suggestion
			'.jpg and jpg|jpeg|jpe mimetypes'    => array(
```
I'd also suggest these additional datasets:
1. A `.jpeg` file with the same value for `$mimes` as this dataset.
2. A `.jpe` file with the same value for `$mimes` as this dataset.
3. A `.JPG` file (uppercase) to ensure the regex in `wp_check_filetype()` is always case-insensitive.

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+				'filename' => 'canola.jpg',
+				'mines'    => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:32]:
```suggestion
				'mimes'    => array(
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+					'jpg|jpeg|jpe' => 'image/jpeg',
+					'gif'          => 'image/gif',
+				),
+				'expected' => array(
+					'ext'  => 'jpg',
+					'type' => 'image/jpeg',
+				),
+			),
+			'badfile'     => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:41]:
```suggestion
			'.XXX and no matching mimetype'      => array(
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+				'filename' => 'canola.XXX',
+				'mines'    => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:43]:
```suggestion
				'mimes'    => array(
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+					'jpg|jpeg|jpe' => 'image/jpeg',
+					'gif'          => 'image/gif',
+				),
+				'expected' => array(
+					'ext'  => false,
+					'type' => false,
+				),
+			),
+			'bad_mines'   => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:52]:
```suggestion
			'.jpg but only gif allowed'          => array(
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+				'filename' => 'canola.jpg',
+				'mines'    => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:54]:
```suggestion
				'mimes'    => array(
```

File: tests/phpunit/tests/functions/wpCheckFiletype.php [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+					'gif' => 'image/gif',
+				),
+				'expected' => array(
+					'ext'  => false,
+					'type' => false,
+				),
+			),
+
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpCheckFiletype.php:62]:
```suggestion
```

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the PR @pbearne!

I've left suggestions to bring the tests up to Core standards and have suggested four additional datasets.

Note that there may be PHPCS array alignment failures if these suggestions are applied directly in GitHub.

**Comment by hellofromtonya** [comment]
Closing in favor of https://github.com/WordPress/wordpress-develop/pull/4158, which was committed.

**Pull request closed without merging**